vmm/src/vmm_config/logger.rs

```diff
@@ -109,6 +113,26 @@ fn default_log_options() -> Value {
     Value::Array(vec![])
 }
 
+fn validate_logger_options<'de, D>(d: D) -> std::result::Result<Value, D::Error>
+where
+    D: de::Deserializer<'de>,
+{
+    let options = Value::deserialize(d)?;
+
+    if options == Value::Null {
+        return Ok(default_log_options());
+    }
+
+    if options.is_array() {
+        Ok(options)
```

Comment:
This still doesn't check that options only contains valid elements. We should probably also validate that options is an array of `LogOption`. And since we want this check in place as well I was wondering if we can actually change the type of the `options` field from `Value` to `Vec<LogOption>`. This should simplify the `set_flags` function in the logger.

Reply:
Yes, since I noticed that the set_flags function would do  the validation,  I didn't do it here. If you think it's better to move the validation here and I would update it later.